### ISSUE TITLE
Classes.SuperfluousExceptionNaming.SuperfluousSuffix should not trigger when the name is just Exception

### ISSUE THREAD

**Issue by PeeHaa** [slevomat/coding-standard#552]
Currently the rule `SlevomatCodingStandard.Classes.SuperfluousExceptionNaming.SuperfluousSuffix` is being triggered when having an exception class with a name of `Exception`:

```php
<?php

namespace MyLibrary;

class Exception extends \Exception
{
}
```

> 5 | ERROR | Superfluous suffix "Exception".
>    |       | (SlevomatCodingStandard.Classes.SuperfluousExceptionNaming.SuperfluousSuffix)

This should not happen as it is not a suffix in this case, but the actual exception / class name.

**Comment by PeeHaa** [comment]
FWIW my proposed fix for this issue would be the following change in the [`SuperfluousExceptionNamingSniff` sniff](https://github.com/slevomat/coding-standard/blob/master/SlevomatCodingStandard/Sniffs/Classes/SuperfluousExceptionNamingSniff.php#L35-L50):

```php
public function process(File $phpcsFile, $classPointer): void
{
    $className = ClassHelper::getName($phpcsFile, $classPointer);

    $previousPointer = TokenHelper::findPreviousEffective($phpcsFile, $classPointer - 1);

    if ($phpcsFile->getTokens()[$previousPointer]['code'] === T_ABSTRACT) {
        return;
    }

    $normalizedClassName = strtolower($className);

    $suffix = substr($className, -9);
    if ($normalizedClassName === 'exception' || $suffix !== 'exception') {
        return;
    }

    $phpcsFile->addError(sprintf('Superfluous suffix "%s".', $suffix), $classPointer, self::CODE_SUPERFLUOUS_SUFFIX);
}
```

I would PR the change myself, but I keep hitting issues (like line-endings (easy to fix) and "Allowed memory size") with the build process on my dev machine unrelated to my fix.

Would be great of somebody can pick this up for me. It would also need an extra test for this issue, but that is also really straight forward.

**Comment by kukulich** [comment]
Thank you for your report.

Fixed in https://github.com/slevomat/coding-standard/commit/e06c13eb758388605416b5daf33c8ce923708d44
